x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_table/result_settings_table.scss

```diff
@@ -0,0 +1,30 @@
+.resultSettingsTable {
+  table-layout: auto;
+
+  &__columnLabels {
+    .euiTableHeaderCell {
+      border-bottom: none;
+      padding-bottom: $euiSizeM;
+    }
+  }
+
+  .euiTableRow:last-of-type {
+    .euiTableRowCell, .euiTableRowCellCheckbox {
+      // Because we are using the large border-top as a way of spacing, and table borders
+      // are collapsed, these tables do not have a bottom border. The exception to that is
+      // if the tbody is the last tbody of the table. To make it consistent, we explicitly
+      // disable all bottom borders here
+      border-bottom: none;
+    }
+  }
+
+  tbody + thead {
```

Comment:
This table is a little bit odd. We have multiple `tbody` elements inside of a single table, which gives the appearance of separate tables even though it is a single table. The large border style below is the only way I could work out to add vertical space between the table sections.

I actually tried to convert this code to use wholly separate tables, but it was really challenging to get all of the columns within the tables to line up consistently.

Reply:
This probably wildly out of scope - I'm definitely not sure and would love your more informed opinion. On a scale of 1-10, how feasible do you think it would be to convert this super custom EuiTable with individual components to an `EuiBasicTable` instead? Definitely not suggesting this as change request to this PR, would just be generally curious about your thoughts for if/when we did refactor this view some day in the future.

Reply:
Oh, wait, sorry, I think I see what's going on.... (with the "separate" non-text fields and the disabled fields tables).

Unfortunately I'm not sure our current implementation is semantically valid HTML. You're only supposed to have one `thead` and `tfoot` element per `table`, although weirdly enough you can have multiple `tbodies` ([whatwg source](https://html.spec.whatwg.org/multipage/tables.html#the-table-element)).

That being said, I'm fine leaving this as-is for the sake of migration and I definitely don't think we need to fix this now. However that does strengthen my opinion that we should definitely come back to this and generally overhaul this UI later 🤔

Reply:
Actually, [per MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tbody#multiple_bodies) I think we can somewhat quickly fix the multiple `thead` thing by simply using `<tr>` and `<th>` instead of a `<thead>`. I'll see if I can make some quick code suggestions for that.

Reply:
Ahhhh, I didn't see that about multiple theads. I might be able to address that easily by using their suggestion:

> because of that, you need to use a <tr> filled with <th> elements to create headers within each <tbody>.

And I don't think converting to a `EuiBasicTable` is possible just because of the sub sections within the table.